### ISSUE TITLE
ansible.builtin.systemd_service not recognized

### ISSUE THREAD

**Issue by commodis** [ansible/ansible-lint#4185]
##### Summary

`ansible-lint` does not recognize the builtin module `ansible.builtin.systemd_service` used for `systemd`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version

ansible-lint 6.8.2 using ansible 2.13.5
```

this is not the latest version, but I am using ansible-lint through `Docker` and the image not updated yet. But I have checked the changelog and I did not find anything related in the [ansible-lint releases](https://github.com/ansible/ansible-lint/releases).

##### STEPS TO REPRODUCE

In order to use `daemon_reload` it is necessary to use the module `ansible.builtin.systemd_service` (`service` does not offer a daemon-reload):

```yml
- name: systemctl restart ssh.socket
  ansible.builtin.systemd_service:
    # restart daemon before ssh.socket is restarted
    daemon_reload: true
    name: ssh.socket
    state: restarted
```

but `ansible-lint` trips on this with this message:

```console
d=975254;https://ansible-lint.readthedocs.io/rules/syntax-check/1;91msyntax-check[specific]: couldn't resolve module/action 'ansible.builtin.systemd_service'. This often indicates a misspelling, missing collection, or incorrect module path.
```

I also tried searching the codebase for `systemd_service` but did not match anything:

* neither for fqcn https://github.com/search?q=repo%3Aansible%2Fansible-lint+ansible.builtin.systemd_service&type=code
* nor the simple https://github.com/search?q=repo%3Aansible%2Fansible-lint+systemd_service&type=code

**Comment by commodis** [comment]
I will try again with the latest version - thanks for the feedback.
